### ISSUE TITLE
Show more ignores search context

### ISSUE THREAD

**Issue by grtjn** [joemfb/ml-search-ng#83]
If you enable show-more, select some facet, and then click 'show more', that will return 5 more facet values with wrong frequency count. It looks like it pulls values from /v1/values without sending along the structured query that is needed to get the correct values/counts..

**Comment by grtjn** [comment]
I'm using ml-search-ng#0.2.2, and ml-common-ng#0.2.2

**Comment by joemfb** [comment]
Are you using additional-query in your search options? I'm not sure if that's respected by /v1/values ...

**Comment by grtjn** [comment]
Only in stored options, and given the scale of difference that wouldn't explain it. It really is the selected facets that should make most difference..

**Comment by joemfb** [comment]
I'll try to reproduce when I can, but, in the meantime, can you comment with the queries being sent for both /v1/search and /v1/values?

**Comment by grtjn** [comment]
The search is a GET /v1/search, and has a structuredQuery parameter containing:
```json
{"query":{"queries":[{"and-query":{"queries":[{"collection-constraint-query":{"constraint-name":"Collection","uri":["investigations"]}}]}},{"operator-state":{"operator-name":"results","state-name":"compact"}}]}}
```
That results in these 5 'Make' facet values:
```
 Ford (6646) 
 Dodge (3897) 
 Chevrolet (3683) 
 Honda (3290) 
 Bmw (2670)
```

Clicking the `show more` triggers a POST /v1/values/Make with this request body:
```json
{"search":{"query":{"queries":[{"and-query":{"queries":[{"collection-constraint-query":{"constraint-name":"Collection","uri":["investigations"]}}]}},{"operator-state":{"operator-name":"results","state-name":"compact"}}]},"qtext":"","options":{"constraint":[{"name":"Make","range":{"type":"xs:string","facet":true,"collation":"http://marklogic.com/collation/codepoint","facet-option":["limit=5","frequency-order","descending"],"field":{"name":"Make"}}}],"values":[{"name":"Make","range":{"type":"xs:string","facet":true,"collation":"http://marklogic.com/collation/codepoint","facet-option":["limit=5","frequency-order","descending"],"field":{"name":"Make"}},"values-option":["limit=5","frequency-order","descending"]}]}}}
```

That returns these additional facet values:
```
 Gmc (64544) 
 Jeep (62848) 
 Nissan (60968) 
 Chrysler (50259) 
 Pontiac (43919) 
```

**Comment by grtjn** [comment]
So interestingly, the structuredQuery is included, but for some reason not applied. The docs of POST /v1/values/.. seems to suggest that is supposed to work..

**Comment by joemfb** [comment]
What do you mean by the "structuredQuery is not applied"? Are the initial facet results inconsistent with your data? Maybe you could just do some testing in qconsole?

    for $val in cts:values(
      cts:field-reference("Make", "collation=http://marklogic.com/collation/codepoint"),
      (),
      ("limit=5", "frequency-order", "descending"),
      cts:collection-query("investigations"))
    return $val || "-" || cts:frequency($val)

And then run it again with your additional-query from the stored options: `cts:and-query((cts:query( $additional-query ), cts:collection-query("investigations")))`...

**Comment by grtjn** [comment]
I was talking about the call to /v1/values. It does contain the same structured query as the call to /v1/search, but where /v1/search gives me facet counts that are significantly lower than with an empty search, the next 5 returned by /v1/values are much higher, as if the structured query is not applied..

**Comment by grtjn** [comment]
For the sake of completeness, your code (with two minor tweaks) returns:
```
GMC-2167
RICON-2059
TOYOTA-1682
MAZDA-1625
CHRYSLER-1506
```
E.g. much lower numbers..

**Comment by joemfb** [comment]
That's weird. Did you try it with and without your additional-query? Same results?

BTW, could this be an issue of item vs fragment frequency? I think facets in the Search API respect the defaults of the lexicon APIs (fragment-frequency), but maybe there's a mismatch somewhere?

**Comment by joemfb** [comment]
Confirmed bug; options param not passed to /v1/values, so queries against non-existing constraints are silently ignored (!!!).

**Comment by joemfb** [comment]
Fixed in latest release: v0.2.3.
